### PR TITLE
[Testcase Refactoring]Add HardwareClassification label to test_torchbind.py

### PR DESCRIPTION
**Pull request by chenm-123**
- Add hw_classification = HardwareClassification.GENERIC to TestTorchbind

Metadata-only change. No functional impact.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: test/inductor/test_torchbind.py
Comment (wyb2939) [inline test/inductor/test_torchbind.py]: This function has the `requires_gpu` decorator, utilizes `GPU_TYPE`, and needs to be split into a separate class, labeled with the `ACCELERATOR` tag

File: test/inductor/test_torchbind.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from torch._inductor.codecache import WritableTempFile
 from torch._inductor.package import package_aoti
 from torch._inductor.test_case import run_tests, TestCase
-from torch.testing._internal.common_utils import skipIfWindows
+from torch.testing._internal.common_utils import HardwareClassification, skipIfWindows
 from torch.testing._internal.inductor_utils import GPU_TYPE, requires_gpu
```
Comment (wyb2939) [inline test/inductor/test_torchbind.py]: GPU_TYPE and requires_gpu are hardcoded and need to be decoupled.

File: test/inductor/test_torchbind.py [hunk 2]
```diff
@@ -441,5 +446,8 @@ def forward(self, x):
         self.assertEqual(result, orig_res)
 
 
+instantiate_device_type_tests(TestTorchbind, globals(), allow_xpu=True)
```
Comment (rogerchen838) [inline test/inductor/test_torchbind.py]: This is blocking because the classification filter must not misclassify accelerator-only coverage.

> AI-assisted analysis: `instantiate_device_type_tests(TestTorchbind, ...)` creates accelerator variants by copying this class attribute, so `TestTorchbindCUDA`/`XPU` remain `GENERIC`. As a result, `--hw-classification GENERIC` includes `test_torchbind_compile_gpu_op_symint_graph_partition`, while `ACCELERATOR` does not select it; the device-agnostic tests are also duplicated across devices. Split the accelerator test into an `ACCELERATOR` device-type class and keep only shared tests in the `GENERIC` class.

**Review by rogerchen838 — CHANGES_REQUESTED** [review]
AI-assisted review; the blocking finding is inline.

**Review by rogerchen838 — APPROVED** [review]
I reviewed the current head against the PyTorch device-generalization requirements.

> AI-assisted review used as supporting analysis; the reviewer verified the following against the current head:
> No blocking issue found after reviewing the current head.

File: test/inductor/test_torchbind.py [hunk 3]
```diff
@@ -441,5 +426,38 @@ def forward(self, x):
...
 
 
+class TestTorchbindAccelerator(TestCase):
+    hw_classification = HardwareClassification.ACCELERATOR
+
+    def setUp(self):
+        super().setUp()
+        init_torchbind_implementations()
+
+    @onlyAccelerator
```
Comment (orangeH25) [inline test/inductor/test_torchbind.py]: `@onlyAccelerator` is redundant here since `except_for="cpu"` already prevents CPU test classes from being instantiated.

File: test/inductor/test_torchbind.py [hunk 4]
```diff
@@ -441,5 +426,38 @@ def forward(self, x):
...
+    @torch._dynamo.config.patch("capture_dynamic_output_shape_ops", True)
+    @torch._inductor.config.patch("graph_partition", True)
+    def test_torchbind_compile_gpu_op_symint_graph_partition(self, device):
```
Comment (orangeH25) [inline test/inductor/test_torchbind.py]: Could we rename `gpu_op` to `accelerator_op`?

File: test/inductor/test_torchbind.py [hunk 5]
```diff
@@ -441,5 +426,38 @@ def forward(self, x):
...
+        class M(torch.nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.attr = torch.classes._TorchScriptTesting._Foo(2, 3)
+
+            def forward(self, x):
+                a = torch.ops._TorchScriptTesting.takes_foo_tensor_return(self.attr, x)
+                a_cuda = a.to(device=device)
+                return a_cuda + 1
```
Comment (orangeH25) [inline test/inductor/test_torchbind.py]: Ditto.

File: test/inductor/test_torchbind.py [hunk 6]
```diff
@@ -441,5 +426,38 @@ def forward(self, x):
...
+
+        m = M()
+        inputs = (torch.ones(2, 3),)
+        orig_res = m(*inputs)
+        new_res = torch.compile(m, backend="inductor")(*inputs)
+        self.assertTrue(torch.allclose(orig_res, new_res))
+
+
+instantiate_device_type_tests(
+    TestTorchbindAccelerator, globals(), allow_xpu=True, except_for="cpu"
```
Comment (orangeH25) [inline test/inductor/test_torchbind.py]: Should we add `allow_mps=True` here? Otherwise, this would be inconsistent with the original `@requires_gpu()` semantics.

**Review by orangeH25 — COMMENTED** [review]
Hi @chenm-123 , I left a few comments. Please take a look when you have a chance.

File: test/inductor/test_torchbind.py (hunk 3 above)
Comment (chenm-123) [reply test/inductor/test_torchbind.py]: fixed

File: test/inductor/test_torchbind.py (hunk 4 above)
Comment (chenm-123) [reply test/inductor/test_torchbind.py]: fixed

File: test/inductor/test_torchbind.py (hunk 5 above)
Comment (chenm-123) [reply test/inductor/test_torchbind.py]: fixed

File: test/inductor/test_torchbind.py (hunk 6 above)
Comment (chenm-123) [reply test/inductor/test_torchbind.py]: fixed

**Review by orangeH25 — APPROVED** [review]

**Review by FuDdd — APPROVED** [review]
